api/src/main/java/org/openmrs/api/impl/AdministrationServiceImpl.java

```diff
@@ -150,7 +152,54 @@ public String getGlobalProperty(String propertyName) throws APIException {
 			return null;
 		}
 		
-		return dao.getGlobalProperty(propertyName);
+		GlobalProperty gp = dao.getGlobalPropertyObject(propertyName);
+		if (gp != null && StringUtils.isNotBlank(gp.getPropertyValue())) {
```

Comment:
Even if the global property has an empty value(isBlank), we should still throw an exception for thoses without the required privilege.